lib/Model/Importer/DebugInfo/PDBImporter.cpp

```diff
@@ -0,0 +1,1093 @@
+/// \file PDBImporter.cpp
+/// \brief
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "llvm/DebugInfo/CodeView/CVSymbolVisitor.h"
+#include "llvm/DebugInfo/CodeView/LazyRandomTypeCollection.h"
+#include "llvm/DebugInfo/CodeView/SymbolDeserializer.h"
+#include "llvm/DebugInfo/CodeView/SymbolRecord.h"
+#include "llvm/DebugInfo/CodeView/SymbolVisitorCallbackPipeline.h"
+#include "llvm/DebugInfo/CodeView/SymbolVisitorCallbacks.h"
+#include "llvm/DebugInfo/CodeView/TypeDumpVisitor.h"
+#include "llvm/DebugInfo/PDB/Native/DbiStream.h"
+#include "llvm/DebugInfo/PDB/Native/GlobalsStream.h"
+#include "llvm/DebugInfo/PDB/Native/InfoStream.h"
+#include "llvm/DebugInfo/PDB/Native/InputFile.h"
+#include "llvm/DebugInfo/PDB/Native/ModuleDebugStream.h"
+#include "llvm/DebugInfo/PDB/Native/NativeSession.h"
+#include "llvm/DebugInfo/PDB/Native/PDBFile.h"
+#include "llvm/DebugInfo/PDB/Native/SymbolStream.h"
+#include "llvm/DebugInfo/PDB/PDB.h"
+
+#include "revng/Model/Binary.h"
+#include "revng/Model/Importer/DebugInfo/PDBImporter.h"
+#include "revng/Model/Pass/AllPasses.h"
+#include "revng/Model/Processing.h"
+#include "revng/Model/QualifiedType.h"
+#include "revng/Model/Type.h"
+#include "revng/Support/Assert.h"
+#include "revng/Support/Debug.h"
+#include "revng/Support/MetaAddress.h"
+
+using namespace llvm;
+using namespace llvm::codeview;
+using namespace llvm::object;
+using namespace llvm::pdb;
+
+static Logger<> DILogger("pdb-importer");
+
+namespace {
+class PDBImporterImpl {
+private:
+  PDBImporter &Importer;
+  DenseMap<TypeIndex, model::TypePath> ProcessedTypes;
+
+public:
+  PDBImporterImpl(PDBImporter &Importer) : Importer(Importer) {}
+  void run(NativeSession &Session);
+
+private:
+  void populateTypes();
+  void populateSymbolsWithTypes(NativeSession &Session);
+};
+
+class PDBImporterTypeVisitor : public TypeVisitorCallbacks {
```

Comment:
AFAIU the implementation of the importer is in good part in this class, but the methods are spread out, which makes it a bit difficult to follow.
Can you add a comment to this class explaining from an high level perspective how it works? A sort of reading guide?

Reply:
Done for both `PDBImporterTypeVisitor` and `PDBImporterSymbolVisitor`.